sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs

```diff
@@ -13,69 +13,50 @@ namespace Azure.Security.KeyVault
+        /// <inheritdoc cref="BearerTokenChallengeAuthenticationPolicy.AuthenticateRequestOnChallengeAsync(HttpMessage, bool)" />
+        protected override async Task AuthenticateRequestAsync(HttpMessage message, bool async)
         {
             if (message.Request.Uri.Scheme != Uri.UriSchemeHttps)
             {
                 throw new InvalidOperationException("Bearer token authentication is not permitted for non TLS protected (https) endpoints.");
             }
 
-            // if this policy doesn't have _scope cached try to get it from the static challenge cache.
-            if (_scope != null)
+            // If this policy doesn't have _scope cached try to get it from the static challenge cache.
+            if (_scope == null)
             {
-                return;
+                string authority = GetRequestAuthority(message.Request);
+                _scopeCache.TryGetValue(authority, out _scope);
             }
 
-            string authority = GetRequestAuthority(message.Request);
-            _scopeCache.TryGetValue(authority, out _scope);
-
-            if (_scope == null)
+            if (_scope != null)
             {
-                // The body is removed from the initial request because Key Vault supports other authentication schemes which also protect the body of the request.
-                // As a result, before we know the auth scheme we need to avoid sending an unprotected body to Key Vault.
-                // We don't currently support this enhanced auth scheme in the SDK but we still don't want to send any unprotected data to vaults which require it.
+                // We fetched the scope from the cache, but we have not initialized the Scopes in the base yet.
+                var context = new TokenRequestContext(_scope.Scopes, message.Request.ClientRequestId);
+                await SetAuthorizationHeader(message, context, async).ConfigureAwait(false);
+                return;
+            }
 
-                RequestContent originalContent = message.Request.Content;
-                message.Request.Content = null;
+            // The body is removed from the initial request because Key Vault supports other authentication schemes which also protect the body of the request.
+            // As a result, before we know the auth scheme we need to avoid sending an unprotected body to Key Vault.
+            // We don't currently support this enhanced auth scheme in the SDK but we still don't want to send any unprotected data to vaults which require it.
 
-                if (async)
-                {
-                    await ProcessNextAsync(message, pipeline).ConfigureAwait(false);
-                }
-                else
-                {
-                    ProcessNext(message, pipeline);
-                }
+            message.SetProperty(KeyVaultStashedContentKey, message.Request.Content);
```

Comment:
How are these properties stored? They aren't logged or anything, are they? And what's the point of storing it since a retry will send it again? Or is the retry happening from within this policy now?